server/src/test/java/org/apache/kafka/server/AssignmentsManagerTest.java

```diff
@@ -14,451 +14,506 @@
-        topicNames.put(TOPIC_1, "TOPIC_1");
-        topicNames.put(TOPIC_2, "TOPIC_2");
-        Map<Uuid, String> dirPaths = new HashMap<>();
-        dirPaths.put(DIR_1, "DIR_1");
-        dirPaths.put(DIR_2, "DIR_2");
-        dirPaths.put(DIR_3, "DIR_3");
-        manager = new AssignmentsManager(time, channelManager, 8, () -> 100L,
-                id -> Optional.ofNullable(dirPaths.get(id)), id -> Optional.ofNullable(topicNames.get(id)));
+    private static final MetadataImage TEST_IMAGE;
+
+    static {
+        MetadataDelta delta = new MetadataDelta.Builder().
+            setImage(MetadataImage.EMPTY).
+            build();
+        delta.replay(new FeatureLevelRecord().
+            setName(MetadataVersion.FEATURE_NAME).
+            setFeatureLevel(MetadataVersion.IBP_3_8_IV0.featureLevel()));
+        delta.replay(new RegisterBrokerRecord().
+            setBrokerId(0).
+            setIncarnationId(Uuid.fromString("JJsH6zB0R7eKbr0Sy49ULw")).
+            setBrokerEpoch(123));
+        delta.replay(new RegisterBrokerRecord().
+            setBrokerId(1).
+            setIncarnationId(Uuid.fromString("DtnWclXyQ4qNDvL97JlnvQ")).
+            setBrokerEpoch(456));
+        delta.replay(new RegisterBrokerRecord().
+            setBrokerId(2).
+            setIncarnationId(Uuid.fromString("UFa_RKgLR4mxEXyquEPEmg")).
+            setBrokerEpoch(789));
+        delta.replay(new RegisterBrokerRecord().
+            setBrokerId(3).
+            setIncarnationId(Uuid.fromString("jj-cnHYASAmb_H9JR6nmtQ")).
+            setBrokerEpoch(987));
+        delta.replay(new TopicRecord().
+            setName("foo").
+            setTopicId(TOPIC_1));
+        delta.replay(new PartitionRecord().
+            setPartitionId(0).
+            setTopicId(TOPIC_1).
+            setReplicas(Arrays.asList(0, 1, 2)).
+            setIsr(Arrays.asList(0, 1, 2)).
+            setLeader(1));
+        delta.replay(new PartitionRecord().
+            setPartitionId(1).
+            setTopicId(TOPIC_1).
+            setReplicas(Arrays.asList(1, 2, 3)).
+            setIsr(Arrays.asList(1, 2, 3)).
+            setLeader(1));
+        delta.replay(new TopicRecord().
+            setName("bar").
+            setTopicId(TOPIC_2));
+        delta.replay(new PartitionRecord().
+            setPartitionId(0).
+            setTopicId(TOPIC_2).
+            setReplicas(Arrays.asList(0, 3, 2)).
+            setIsr(Arrays.asList(0, 3, 2)).
+            setLeader(1));
+        delta.replay(new PartitionRecord().
+            setPartitionId(1).
+            setTopicId(TOPIC_2).
+            setReplicas(Arrays.asList(1, 2, 3)).
+            setIsr(Arrays.asList(2)).
+            setLeader(2));
+        delta.replay(new PartitionRecord().
+            setPartitionId(2).
+            setTopicId(TOPIC_2).
+            setReplicas(Arrays.asList(3, 2, 1)).
+            setIsr(Arrays.asList(3, 2, 1)).
+            setLeader(3));
```

Comment:
I think these aren't necessary?

Reply:
I wanted to have one or two partitions that weren't used just to make it a bit more "realistic"